### PR TITLE
Fix issue with downloading images from Digikeys CDN provider Akamai

### LINKED ISSUE #5: Failure to download image from Digikey
I have recently setup inventree and found this very helpful tool.  Been playing with it today and all is working great, except downloading of images from digikey.  I see the following error, but on inspection I see that it's failing to download the image.  It's getting a 403 error code.  I have downloaded the image from a browser with no issues.  I thought maybe I had some cached authentication in the browser cache.  So I tried a browser that I've not used to log into digikey and it also downloaded just fine.  So a bit puzzled why the script would fail with a 403.  

```
Trying URL https://mm.digikey.com/Volume0/opasdata/d220001/medias/images/5565/MFG_PI_5_HERO.jpg
<Response [403]>
ERROR: Request code is 403
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.9/3.9.16/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/Cellar/python@3.9/3.9.16/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/bpayne/wip/inventory/Digikey-Inventree-Integration/inventree_digikey/__main__.py", line 23, in <module>
    add_digikey_part(dkpart)
  File "/Users/bpayne/wip/inventory/Digikey-Inventree-Integration/inventree_digikey/Inventree.py", line 27, in add_digikey_part
    inv_part = create_inventree_part(dkpart)
  File "/Users/bpayne/wip/inventory/Digikey-Inventree-Integration/inventree_digikey/Inventree.py", line 81, in create_inventree_part
    upload_picture(dkpart, part)
  File "/Users/bpayne/wip/inventory/Digikey-Inventree-Integration/inventree_digikey/Inventree.py", line 121, in upload_picture
    invPart.uploadImage(img_file)
  File "/Users/bpayne/wip/inventory/venv/lib/python3.9/site-packages/inventree/base.py", line 543, in uploadImage
    raise TypeError(f"uploadImage called with invalid image: '{image}'")
TypeError: uploadImage called with invalid image: '-1'
```

### PR DESCRIPTION
**Pull request by benpayne**
The requests api is being blocked by Akamai and returning 403 errors.  Loading these same URLs in a browser works just fine and wget will download them.  Curl will not, at first I thought HTTP headers, but the http.client API works with not header modifications at all.  I would guess it's something about the SSL library used by these tools, but that's just a guess.

TLDR - Changes to using http.client instead of requests api and downloading images now works

Closes #5

### CODE REVIEW

**Comment by benpayne** [comment]
Rebased the code to work on the latest version of the code.

**Review by EUdds — APPROVED** [review]
